### PR TITLE
Lenient bytes deserialization

### LINKED ISSUE #2023: No longer able to send when using parity
crossed posted from mist issue log as unsure which causes the issue:

https://github.com/ethereum/mist/issues/1133

on latest parity release 

System information

Version: `0.8.2`
OS & Version: windows 10
Node type: parity
I am no longer able to send any ethers using parity synced to the latest block. it gives me the error msg "We can't estimate your gas usage properly" in red font bottom right of screen even when i am not sending full amount. When i checked my etc parity mist on another pc, the same error msg is reflected. When I use an earlier version 0.8.1, same message happens. Only when switching back to using geth does this error message disappear and i can see the estimated recommended gas fee calculated.

### PR DESCRIPTION
**Pull request by tomusdrw**
Fixes #2023
Possibly related: #2004

### CODE REVIEW

**Comment by rphmeier** [comment]
I don't think we should stop adhering to the spec explicitly. If the mist/geth teams diverge from it then they have a bug in their software. We could do this in `--geth` mode though.

**Comment by tomusdrw** [comment]
Spec [here](https://github.com/ethereum/wiki/wiki/JSON-RPC#output-hex-values) defines how Output values should be formatted.
We also permit 0-padded quantities as inputs (currently we even output such https://github.com/ethcore/parity/issues/2032)

**Comment by fjl** [comment]
Please note that Geth will start being stricter in the 1.5 release.

**Comment by gavofyork** [comment]
> it actually never defined input parameter format 

It is perfectly well defined. From https://github.com/ethcore/parity/wiki/JSONRPC :

> When encoding UNFORMATTED DATA (byte arrays, account addresses, hashes, bytecode arrays): encode as hex, prefix with "0x", two hex digits per byte
> ...
> WRONG: 004200 (must be prefixed 0x)

This is not specific to one side or the other; encoding/decoding refers to both sides of the RPC call, since we're talking about all data which is transported,.

**Comment by rphmeier** [comment]
It's well defined in our fork of the spec but not in the Ethereum/wiki version, although I made a tiny patch there to clarify as well.

**Comment by gavofyork** [comment]
this should be reverted ASAP, but seems that for now, it's the least bad option

File: rpc/src/v1/types/bytes.rs [hunk 1]
```diff
@@ -70,7 +70,13 @@ impl Visitor for BytesVisitor {
...
 	fn visit_str<E>(&mut self, value: &str) -> Result<Self::Value, E> where E: Error {
-		if value.len() >= 2 && &value[0..2] == "0x" && value.len() & 1 == 0 {
+		if value.is_empty() {
+			warn!(
+				target: "deprecated",
+				"Deserializing empty string as empty bytes. This is a non-standard behaviour that will be removed in future versions. Please update your code to send `0x` instead!"
+			);
+			Ok(Bytes::new(Vec::new()))
+		} else if value.len() >= 2 && &value[0..2] == "0x" && value.len() & 1 == 0 {
 			Ok(Bytes::new(FromHex::from_hex(&value[2..]).unwrap_or_else(|_| vec![])))
```
Comment (debris) [inline rpc/src/v1/types/bytes.rs]: unrelated

`unwrap_or_else(|_| vec![])` part is really interesting here. Is it really what geth is doing? It should be an error imo

**Pull request merged**